From 40e76307c1ee1ebe4e0142d623071d0f768f2450 Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Wed, 9 Jan 2019 13:01:01 +0000
Subject: [PATCH] egl_viv: cast EGLNative types from vivante functions

All vivante functions return void* even when the actual native type
(EGLNativeDisplayType, EGLNativeWindowType, etc.) might be typedef'ed
as something else, as they do when the CFLAGS from egl.pc are now used.
---
 .../deviceintegration/eglfs_viv/qeglfsvivintegration.cpp      | 4 ++--
 .../deviceintegration/eglfs_viv_wl/qeglfsvivwlintegration.cpp | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_viv/qeglfsvivintegration.cpp b/src/plugins/platforms/eglfs/deviceintegration/eglfs_viv/qeglfsvivintegration.cpp
index 763a4a462b..2fc076ad0c 100644
--- a/src/plugins/platforms/eglfs/deviceintegration/eglfs_viv/qeglfsvivintegration.cpp
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_viv/qeglfsvivintegration.cpp
@@ -65,7 +65,7 @@ void QEglFSVivIntegration::platformInit()
     VivanteInit();
     mNativeDisplay = fbGetDisplay();
 #else
-    mNativeDisplay = fbGetDisplayByIndex(framebufferIndex());
+    mNativeDisplay = static_cast<EGLNativeDisplayType>(fbGetDisplayByIndex(framebufferIndex()));
 #endif
 
     fbGetDisplayGeometry(mNativeDisplay, &width, &height);
@@ -88,7 +88,7 @@ EGLNativeWindowType QEglFSVivIntegration::createNativeWindow(QPlatformWindow *wi
     Q_UNUSED(window)
     Q_UNUSED(format)
 
-    EGLNativeWindowType eglWindow = fbCreateWindow(mNativeDisplay, 0, 0, size.width(), size.height());
+    EGLNativeWindowType eglWindow = static_cast<EGLNativeWindowType>(fbCreateWindow(mNativeDisplay, 0, 0, size.width(), size.height()));
     return eglWindow;
 }
 
diff --git a/src/plugins/platforms/eglfs/deviceintegration/eglfs_viv_wl/qeglfsvivwlintegration.cpp b/src/plugins/platforms/eglfs/deviceintegration/eglfs_viv_wl/qeglfsvivwlintegration.cpp
index 61e2f17766..3bdae239cd 100644
--- a/src/plugins/platforms/eglfs/deviceintegration/eglfs_viv_wl/qeglfsvivwlintegration.cpp
+++ b/src/plugins/platforms/eglfs/deviceintegration/eglfs_viv_wl/qeglfsvivwlintegration.cpp
@@ -60,7 +60,7 @@ void QEglFSVivWaylandIntegration::platformInit()
     }
 
     mWaylandDisplay = wl_display_create();
-    mNativeDisplay = fbGetDisplay(mWaylandDisplay);
+    mNativeDisplay = static_cast<EGLNativeDisplayType>(fbGetDisplay(mWaylandDisplay));
     fbGetDisplayGeometry(mNativeDisplay, &width, &height);
     mScreenSize.setHeight(height);
     mScreenSize.setWidth(width);
@@ -81,7 +81,7 @@ EGLNativeWindowType QEglFSVivWaylandIntegration::createNativeWindow(QPlatformWin
     Q_UNUSED(window)
     Q_UNUSED(format)
 
-    EGLNativeWindowType eglWindow = fbCreateWindow(mNativeDisplay, 0, 0, size.width(), size.height());
+    EGLNativeWindowType eglWindow = static_cast<EGLNativeWindowType>(fbCreateWindow(mNativeDisplay, 0, 0, size.width(), size.height()));
     return eglWindow;
 }
 
